### ISSUE TITLE
Summary command crashing on files with Discriminator objects

### ISSUE THREAD

**Issue by myprivatealaska** [pb33f/openapi-changes#83]
Hello! The program crashes when I'm running the **summary** command on a file containing a Discriminator object. Here's the inputs to reproduce the error:

`original.yml`:
```
openapi: "3.0.1"
components:
  schemas:
    Error:
      properties:
        code:
          description: Error code
          type: string
      type: object
      discriminator:
        propertyName: code
        mapping:
          authorization: "#/components/schemas/AuthorizationError"
          error: "#/components/schemas/GenericError"
      oneOf:
      - "$ref": "#/components/schemas/AuthorizationError"
      - "$ref": "#/components/schemas/GenericError"
    GenericError:
      properties:
        code:
          description: Error code
          type: string
      type: object
    AuthorizationError:
      properties:
        code:
          description: Error code
          example: authorization
          type: string
      type: object
```

`modified.yml`:

```
openapi: 3.0.1
components:
  schemas:
    Error:
      properties:
        code:
          description: Error code
          type: string
      type: object
      discriminator:
        propertyName: code
        mapping:
          authentication: "#/components/schemas/AuthenticationError"
          authorization: "#/components/schemas/AuthorizationError"
          error: "#/components/schemas/GenericError"
      oneOf:
        - "$ref": "#/components/schemas/AuthenticationError"
        - "$ref": "#/components/schemas/AuthorizationError"
        - "$ref": "#/components/schemas/GenericError"
    GenericError:
      properties:
        code:
          description: Error code
          type: string
      type: object
    AuthorizationError:
      properties:
        code:
          description: Error code
          example: authorization
          type: string
      type: object
    AuthenticationError:
      properties:
        code:
          description: Error code
          example: authentication
          type: string
      type: object
```

`openapi-changes summary original.yml modified.yml` throws the following error:

```
panic: interface conversion: interface {} is model.DiscriminatorChanges, not model.SchemaChanges

goroutine 1 [running]:
github.com/pb33f/openapi-changes/builder.exploreTreeObject(0xc0002b78f0, {0xbde480?, 0xc000122d20?})
        D:/a/openapi-changes/openapi-changes/builder/tree.go:213 +0x4094
github.com/pb33f/openapi-changes/builder.BuildTreeMapNode(0xc0002b77a0, {0xbb45a0?, 0xc0005a4048?, 0xb7bf44?})
        D:/a/openapi-changes/openapi-changes/builder/tree.go:350 +0xa5
github.com/pb33f/openapi-changes/builder.exploreTreeObject(0xc0002b77a0, {0xbdf080?, 0xc0005a4040?})
        D:/a/openapi-changes/openapi-changes/builder/tree.go:236 +0x2010
github.com/pb33f/openapi-changes/builder.DigIntoTreeNode[...](0xc000189f10, {0xbdf080, 0xc00052c070?, 0x10e13c8?}, {0xc00035ab90?, 0xa}, 0x55, 0x10)
        D:/a/openapi-changes/openapi-changes/builder/tree.go:300 +0x2ea
github.com/pb33f/openapi-changes/builder.exploreTreeObject(0xc000189f10, {0xbdf480?, 0xc00052c000?})
        D:/a/openapi-changes/openapi-changes/builder/tree.go:141 +0x3205
github.com/pb33f/openapi-changes/builder.BuildTree({0xbdf480, 0xc00052c000})
        D:/a/openapi-changes/openapi-changes/builder/tree.go:35 +0x9d
github.com/pb33f/openapi-changes/cmd.printSummaryDetails({0xc00081fbf0, 0x2, 0xc000539a28?})
        D:/a/openapi-changes/openapi-changes/cmd/summary.go:404 +0x2bd
github.com/pb33f/openapi-changes/cmd.runLeftRightSummary({0x7ffdc327331c, 0x12}, {0x7ffdc327332f, 0x16}, 0x4?, 0x1000330?, {0x0, 0x0}, 0x0?)
        D:/a/openapi-changes/openapi-changes/cmd/summary.go:326 +0x8c5
github.com/pb33f/openapi-changes/cmd.GetSummaryCommand.func1(0xc0002a2100?, {0xc0001d26a0, 0x2, 0xd209f0?})
        D:/a/openapi-changes/openapi-changes/cmd/summary.go:228 +0x975
github.com/spf13/cobra.(*Command).execute(0xc0002b8300, {0xc0001d2640, 0x2, 0x2})
        C:/Users/runneradmin/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:983 +0xabc
github.com/spf13/cobra.(*Command).ExecuteC(0x173f2e0)
        C:/Users/runneradmin/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:1115 +0x3ff
github.com/spf13/cobra.(*Command).Execute(...)
        C:/Users/runneradmin/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:1039
github.com/pb33f/openapi-changes/cmd.Execute({0x10c7fb0?, 0x19?}, {0x10d1e60?, 0xd2ed44?}, {0xc0001926a0?, 0xc0000061a0?})
        D:/a/openapi-changes/openapi-changes/cmd/root.go:51 +0xb0
main.main()
        D:/a/openapi-changes/openapi-changes/openapi-changes.go:29 +0x158
```

**Comment by daveshanley** [comment]
Thank you, will fix.

**Comment by daveshanley** [comment]
This has been resolved in `v0.0.52`

Here is the output showing these examples:

![Screenshot 2024-01-05 at 8 58 00 AM](https://github.com/pb33f/openapi-changes/assets/187345/6c4400b3-bf43-4f87-bbdb-2a6834686301)
